--- rspamd-1.8.1-silent-logger/src/plugins/lua/once_received.lua	2018-11-10 16:28:22.476332448 +0100
+++ rspamd-1.8.1/src/plugins/lua/once_received.lua	2018-11-10 16:24:39.165058030 +0100
@@ -48,6 +48,6 @@
 
   local function recv_dns_cb(_, to_resolve, results, err)
     if err and (err ~= 'requested record is not found' and err ~= 'no records with this name') then
-      rspamd_logger.errx(task, 'error looking up %s: %s', to_resolve, err)
+      rspamd_logger.warnx(task, 'error looking up %s: %s', to_resolve, err)
       task:insert_result(symbol_rdns_dnsfail, 1.0)
     end
